packages/multichain-network-controller/src/utils.ts

```diff
@@ -130,3 +151,121 @@ export const toMultichainNetworkConfigurationsByChainId = (
     }),
     {},
   );
+
+// TODO: Move this function to @metamask/utils
+/**
+ * Type guard to check if a namespace is a known CAIP namespace.
+ *
+ * @param namespace - The namespace to check
+ * @returns Whether the namespace is a known CAIP namespace
+ */
+function isKnownCaipNamespace(
+  namespace: string,
+): namespace is KnownCaipNamespace {
+  return Object.values<string>(KnownCaipNamespace).includes(namespace);
+}
+
+/**
+ * Validates CAIP-10 account IDs format.
+ *
+ * @param accountIds - Array of account IDs to validate
+ * @throws Error if any account ID is invalid
+ */
+export function assertCaipAccountIds(
+  accountIds: string[],
+): asserts accountIds is CaipAccountId[] {
+  if (!accountIds.length) {
+    throw new Error('At least one account ID is required');
+  }
+
+  const invalidIds = accountIds.filter((id) => {
+    const {
+      address,
+      chain: { namespace },
+    } = parseCaipAccountId(id as CaipAccountId);
```

Comment:
I believe there's an issue here 🤔 we shouldn't use a type-cast here, cause if `id` turns out to not be a `CaipAccountId`, then this will `throw`.

An alternative would be to use `isCaipAccountId` first, so we avoid throwing an error, and `id` would naturally become a `CaipAccountId` (thanks to a type-predicate).

```suggestion
    if (!isCaipAccountId(id)) {
      return false;
    }

    const {
      address,
      chain: { namespace },
    } = parseCaipAccountId(id);
```

WDYT?